### DISCUSSION TITLE
Error setting up build on Linux

### DISCUSSION THREAD

**Discussion by 2br-2b** [ghostery/user-agent-desktop#497] [Q&A]
I've been making more than enough issues, and I've wanted to contribute, but I have the following error when trying to build.

I ran these commands:

```
git clone https://github.com/ghostery/user-agent-desktop/
cd user-agent-desktop
npm ci
```

I got the error at https://gist.github.com/2br-2b/8c5d9f7f0f683f3671ae7a8c844465ce

What is going wrong? I assumed it was an error with the wrong version of python with some feature not included in this version of python, hence the error `platform is not implemented yet`.

I'm running Linux Mint, and it seems to be trying to work with some macos files. I'm guessing that whatever it's trying to do is not supported on Linux and only on Macs and/or Windows.

This will be my first time contributing to Firefox or any of its forks, so I want to make sure I get the environment set up right.

Thank you for your help!

**Comment by remusao** [comment]
Hey,

I think I encountered this issue before on Linux but it was not so clear why it happened. Can you try to run `npm install` instead of `npm ci`?

Best,

**Reply by 2br-2b** [reply]
Hi,

I get a similar error. See https://gist.github.com/2br-2b/63fb5526cb267ff493d7cb86f6e3d75c

**Reply by 2br-2b** [reply]
I get the same error when running `npm install colors`, so I'm guessing this is the problem

**Reply by 2br-2b** [reply]
I guessed that colors is the issue because running `./fern.js use` gives the following error:

```
foo:~/uncompiled/ghostery-browser$ ./fern.js use
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'colors'
Require stack:
- /home/foo/uncompiled/ghostery-browser/fern.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/home/foo/uncompiled/ghostery-browser/fern.js:6:16)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/home/foo/uncompiled/ghostery-browser/fern.js' ]
}
```

**Reply by 2br-2b** [reply]
After some troubleshooting, it appears that the `colors` and `appdmg` dependencies only work on the Darwin platform (developed by Apple). Adding them to the list of optional dependencies lets me run `npm install`  but gives me an error when using fern.js

```
foo:~/gb$ npm install

added 228 packages, and audited 229 packages in 3s

found 0 vulnerabilities

foo:~/gb$ ./fern,js use
  ❯ Setup Firefox 85.0.2
    ✔ Download
    ✖ Extract
      → firefox-85.0.2/.eslintignore
      Git
      Link
      Populate Build Folder
    Setup Addons
    Setup Locales
    Persist Workspace
```

**Reply by remusao** [reply]
Hey, thanks a lot for looking into it further. I think we have found the culprit and I commented on the dedicated issue directly: https://github.com/ghostery/user-agent-desktop/issues/507
